flow/layers/layer.cc

```diff
@@ -96,7 +111,19 @@ Layer::AutoSaveLayer Layer::AutoSaveLayer::Create(
   return Layer::AutoSaveLayer(paint_context, layer_rec, save_mode);
 }
 
+Layer::AutoSaveLayer Layer::AutoSaveLayer::Create(
+    DisplayListBuilder* builder,
+    const SkRect& bounds,
+    const DlPaint* paint,
+    const DlImageFilter* backdrop) {
+  return Layer::AutoSaveLayer(builder, bounds, paint, backdrop);
+}
+
 Layer::AutoSaveLayer::~AutoSaveLayer() {
+  if (type_ == Type::kDisplayList) {
+    dl_builder_->restore();
+    return;
+  }
```

Comment:
In the following lines we always use canvas_ to checkerboard, but this could potentially be a nodes_canvas which would broadcast the checkerboarding to all embedded slices. We should probably always use leaf_canvas here regardless of the "mode" of the SkCanvas saveLayer.